data_structures/trie_beginner.cpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * @file
+ * @brief https://en.wikipedia.org/wiki/Trie
+ * @details Trie is a handy data structure for string matching/searching/prefix
+ * and it's very efficient with both time and space. This implementation is for
+ * beginners where I only used strings with only [LOWER CASE ENGLISH LETTERS]
+ * for simplicity. It has the 3 most basic and most essential functions.
+ * @author [Dipankar](https://github.com/UltimateDip)
+ * @see trie_modern.cpp trie_tree.cpp trie_using_hashmap.cpp for adavanced
+ * usecase
+ */
+
+#include <assert.h>
```

Comment:
C++ includes a header for `assert`.
```suggestion
#include <cassert>
```